returnn/tf/layers/basic.py

```diff
@@ -997,21 +997,29 @@ def __init__(self, start, size, min_size=None, **kwargs):
     start_data = start_data.copy_compatible_to(common_data, check_sparse=False)
     start_t = start_data.placeholder
     if size is None:
-      if min_size is None:
-        min_size = 0
       if seq_lens_data is None:
         assert isinstance(x.batch_shape[x.time_dim_axis], int)
         size_t = x.batch_shape[x.time_dim_axis] - start_t
       else:
         seq_lens_t = seq_lens_data.copy_compatible_to(common_data, check_sparse=False).placeholder
         size_t = seq_lens_t - start_t
-      size = tf.maximum(tf.reduce_max(size_t), min_size)  # scalar
+      size_t = tf.maximum(size_t, min_size or 0)  # must make sure >=0 in any case
+      size = tf.reduce_max(size_t)  # scalar
     elif isinstance(size, LayerBase):
       size_data = size.output.copy_compatible_to(common_data, check_sparse=False)
-      size_t = size_data.placeholder
-      min_size = 0
-      size = tf.maximum(tf.reduce_max(size_t), min_size)  # scalar
+      size_t = size_data.placeholder  # assume already >=0
```

Comment:
Why can we assume here that `size_t >= 0`?

Reply:
I mean that we just do this assumption here. So it is up to the user to not give negative sizes. The user should know about this. When the user knows that they might be negative and should be clipped to 0, `min_size=0` can be explicitly provided.